lib/fastlane/plugin/ddg_apple_automation/actions/asana_find_release_task_action.rb

```diff
@@ -0,0 +1,189 @@
+
+        release_task_url = Helper::DdgAppleAutomationHelper.asana_task_url(release_task_id)
+        release_branch = "release/#{latest_marketing_version}"
+        UI.success("Found #{latest_marketing_version} release task: #{release_task_url}")
+
+        Helper::GitHubActionsHelper.set_output("release_branch", release_branch)
+        Helper::GitHubActionsHelper.set_output("release_task_id", release_task_id)
+        Helper::GitHubActionsHelper.set_output("release_task_url", release_task_url)
+
+        {
+          release_task_id: release_task_id,
+          release_task_url: release_task_url,
+          release_branch: release_branch
+        }
+      end
+
+      def self.find_latest_marketing_version(github_token)
+        client = Octokit::Client.new(access_token: github_token)
+
+        # NOTE: `client.latest_release` returns release marked as "latest", i.e. a public release
+        latest_internal_release = client.releases(@constants[:repo_name], { per_page: 1 }).first
+
+        version = extract_version_from_tag_name(latest_internal_release&.tag_name)
+        if version.to_s.empty?
+          Fastlane::UI.user_error!("Failed to find latest marketing version")
+          return
+        end
+        unless self.validate_semver(version)
+          Fastlane::UI.user_error!("Invalid marketing version: #{version}, expected format: MAJOR.MINOR.PATCH")
+          return
+        end
+        version
+      end
+
+      def self.extract_version_from_tag_name(tag_name)
+        tag_name&.split("-")&.first
+      end
+
+      def self.validate_semver(version)
+        # we only need basic "x.y.z" validation here
+        version.match?(/\A\d+\.\d+\.\d+\z/)
+      end
+
+      def self.find_release_task(version, asana_access_token)
+        # `completed_since=now` returns only incomplete tasks
+        url = Helper::DdgAppleAutomationHelper::ASANA_API_URL + "/sections/#{@constants[:release_section_id]}/tasks?opt_fields=name,created_at&limit=100&completed_since=now"
+
+        release_task_id = nil
+
+        # Go through all tasks in the section (there may be multiple requests in case
+        # there are more than 100 tasks in the section).
+        # Repeat until no more pages are left (next_page.uri is null).
+        loop do
+          response = HTTParty.get(url, headers: { 'Authorization' => "Bearer #{asana_access_token}" })
+
+          unless response.success?
+            UI.user_error!("Failed to fetch release task: (#{response.code} #{response.message})")
+            return
+          end
+          parsed_response = response.parsed_response
+
+          find_hotfix_task_in_response(parsed_response)
+          release_task_id ||= find_release_task_in_response(parsed_response, version)
+
+          url = parsed_response.dig('next_page', 'uri')
+
+          # Don't return as soon as release task is found, as we want to ensure there's no hotfix task
+          break if url.nil?
+        end
+
```

Comment:
I think this can be simplified with asana client, not sure if you want to do it in this pr, or the next?

Reply:
Yes, I should do it. Let me try!